File: go/create_subscription_components.go
// this file is @generated
package meteroid

type CreateSubscriptionComponents struct {
	ExtraComponents []ExtraComponent `json:"extra_components,omitempty"`

	OverriddenComponents []ComponentOverride `json:"overridden_components,omitempty"`

	ParameterizedComponents []ComponentParameterization `json:"parameterized_components,omitempty"`

	RemoveComponents []PriceComponentId `json:"remove_components,omitempty"`
}

File: python/meteroid/models/create_subscription_components.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .component_override import ComponentOverride
from .component_parameterization import ComponentParameterization
from .extra_component import ExtraComponent
from .price_component_id import PriceComponentId


@dataclasses.dataclass
class CreateSubscriptionComponents(BaseModel):
    extra_components: t.Optional[t.List[ExtraComponent]] = None

    overridden_components: t.Optional[t.List[ComponentOverride]] = None

    parameterized_components: t.Optional[t.List[ComponentParameterization]] = None

    remove_components: t.Optional[t.List[PriceComponentId]] = None
